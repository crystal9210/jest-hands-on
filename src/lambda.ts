// src/lambda.ts

import { S3 } from 'aws-sdk';

// Lambdaハンドラー
export const handler = async (event: any): Promise<any> => {
    const s3 = new S3();

    const bucketName = event.bucketName;
    const key = event.key;

    try {
        const data = await s3.getObject({ Bucket: bucketName, Key: key }).promise();
        const body = data.Body?.toString('utf-8') || '';

        // データの処理（例：JSONパース）
        const parsedData = JSON.parse(body);

        return {
        statusCode: 200,
        body: JSON.stringify(parsedData),
        };
    } catch (error: any) {
        return {
        statusCode: 500,
        body: JSON.stringify({ message: error.message }),
        };
    }
};
