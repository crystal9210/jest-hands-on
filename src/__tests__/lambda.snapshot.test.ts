// src/__tests__/lambda.snapshot.test.ts

import AWSMock from 'aws-sdk-mock';
import AWS from 'aws-sdk';
import { handler } from '../lambda';

describe('Lambda Handler Snapshot Tests', () => {
    const mockBucketName = 'test-bucket';
    const mockKey = 'test-key.json';
    const mockData = { message: 'Hello, Snapshot!' };

    beforeAll(() => {
        AWSMock.setSDKInstance(AWS);

        AWSMock.mock('S3', 'getObject', (params: any, callback: Function) => {
        if (params.Bucket === mockBucketName && params.Key === mockKey) {
            callback(null, {
            Body: JSON.stringify(mockData),
            });
        } else {
            callback(new Error('Object not found'));
        }
        });
    });

    afterAll(() => {
        AWSMock.restore('S3');
    });

    test('should match the snapshot for successful fetch', async () => {
        const event = {
        bucketName: mockBucketName,
        key: mockKey,
        };

        const response = await handler(event);
        expect(response).toMatchSnapshot(); // Jestのスナップショットマッチャー
    });

    test('should match the snapshot for fetch error', async () => {
        const event = {
        bucketName: mockBucketName,
        key: 'non-existent-key.json',
        };

        const response = await handler(event);
        expect(response).toMatchSnapshot();
    });
});
