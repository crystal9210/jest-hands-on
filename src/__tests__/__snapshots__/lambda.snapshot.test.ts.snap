// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lambda Handler Snapshot Tests should match the snapshot for fetch error 1`] = `
{
  "body": "{"message":"Object not found"}",
  "statusCode": 500,
}
`;

exports[`Lambda Handler Snapshot Tests should match the snapshot for successful fetch 1`] = `
{
  "body": "{"message":"Hello, Snapshot!"}",
  "statusCode": 200,
}
`;
